bundler/helpers/v2/lib/functions/force_updater.rb

```diff
@@ -53,6 +62,28 @@ def run
                 :update_multiple_dependencies
     alias update_multiple_dependencies? update_multiple_dependencies
 
+    def add_extra_top_level_deps(specs, updates)
+      top_level_deps.each do |dep|
+        name = dep.name
+
+        original_version = original_specs.find { |s| s.name == name }&.version
+        new_version = specs[name].first&.version
+        next unless original_version != new_version
+
+        dependency_of_updated_dep = updates.any? do |updated_dep|
+          new_spec = specs[updated_dep].first
+          next false unless new_spec
+
+          new_spec.dependencies.any? { |d| d.name == name }
+        end
+        next unless dependency_of_updated_dep
```

Comment:
```suggestion
        next unless updates.any? do |updated_dep|
          new_spec = specs[updated_dep].first
          next false unless new_spec

          new_spec.dependencies.any? { |d| d.name == name }
        end
```

Reply:
The name confused me more than the code itself xD
It's probably fine either way really

Reply:
Haha, well if it didn't have the intended effect, let's kill it! 😄